engine_architecture.tex

```diff
@@ -0,0 +1,75 @@
+\section{Game engine architecture}
+New sections should not start with directly a subsection, rather with short section introduction. 
+
+\subsection{Overview of architecture}
+% Opisanie architektury silnika, sposobu komunikacji między komponentami, a także za co one odpowiadają.
+Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam consectetur dignissim urna, a rhoncus dolor venenatis adipiscing. Phasellus viverra aliquam fringilla. Praesent mattis diam id ipsum placerat a laoreet diam tempus. Quisque eu leo ut sapien ultricies gravida ac sed turpis. Donec nibh enim, pretium a sagittis et, varius non nulla. Vivamus enim augue, condimentum eget convallis hendrerit, mattis ut lectus. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Proin semper libero augue. Praesent auctor ligula sit amet quam molestie adipiscing. Mauris vehicula congue mollis. Duis tincidunt interdum purus quis congue. Integer vel orci mauris. Curabitur rutrum tincidunt mi ut elementum. Nunc non massa nunc. Donec ultricies ipsum ac risus sollicitudin porta. Phasellus elementum odio et leo consectetur consectetur. 
+
+Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam consectetur dignissim urna, a rhoncus dolor venenatis adipiscing. Phasellus viverra aliquam fringilla. Praesent mattis diam id ipsum placerat a laoreet diam tempus. Quisque eu leo ut sapien ultricies gravida ac sed turpis. Donec nibh enim, pretium a sagittis et, varius non nulla. Vivamus enim augue, condimentum eget convallis hendrerit, mattis ut lectus. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Proin semper libero augue. Praesent auctor ligula sit amet quam molestie adipiscing. Mauris vehicula congue mollis. Duis tincidunt interdum purus quis congue. Integer vel orci mauris. Curabitur rutrum tincidunt mi ut elementum. Nunc non massa nunc. Donec ultricies ipsum ac risus sollicitudin porta. Phasellus elementum odio et leo consectetur consectetur. 
+
+\subsection{Components of a solution}
+% Opisanie elementów z jakich się składa rozgrywka m.in.: kafelki i ich elementy, gracze, a także uwzględnione rozszerzenia gry Carcassonne.
```

Comment:
Discussion topic: is the description in the comment what we actually intended to include in this section? It's not consistent with the contents of the progress report last semester but it seems consistent with the description that we included in the article draft.
The text that's already here is fine, it's just a matter of whether there isn't something else that we wanted to include that we might otherwise miss.

Reply:
well, I followed the article draft topics, however after checking the semester report I believe my section describes more or less the idea of our solution. I do not think that huge class diagram would fit into the article. Moreover there is no sense describing all structures, but the most crucial ones. The only things I omited were stacks, scoreReport and  logger. Logger could be mentioned in the first paragraph.